File: mario.py
# a program to print complete pyramid with central hole

# importing from the cs50 library
from cs50 import get_int

n = 0
while (n < 1 or n > 8):
    n = get_int("Height: ")

for i in range(0, n, 1):

    # for spaces
    for j in range(n-1, i, -1):
        print(" ", end="")

    # for left side of pyramid
    for k in range(0, i+1, 1):
        print("#", end="")

    # central hole
    print("  ", end="")

    # for right side of pyramid
    for l in range(0, i+1, 1):
        print("#", end="")

    # to start from the new line
    print("")

File: mario.c
// double pyramid mario with hole in the middle
#include <stdio.h>

int main(void)
{
    // getting the height from the user
    int n = 0;
    while (n > 8 || n < 1)
    {
        printf("Height: ");
        scanf("%i", &n);
    }

    // running the loop till height
    for (int i = 0; i < n; i++)
    {
        // for the beginning spaces
        for (int j = n - 1; j > i; j--)
        {
            printf(" ");
        }

        // for the left tiles
        for (int k = 0; k <= i; k++)
        {
            printf("#");
        }

        // the central hole
        printf("  ");

        // for the right tiles
        for (int l = 0; l <= i; l++)
        {
            printf("#");
        }

        printf("\n");
    }
}
